eth_account/hdaccount/deterministic.py

```diff
@@ -0,0 +1,192 @@
+        if node[-1] in ("'", "H"):
+            return HardNode(int(node[:-1]))
+        else:
+            return SoftNode(int(node))
+
+
+class SoftNode(Node):
+    """
+    Soft node (unhardened), where value = index
+    """
+    TAG = ""  # No tag
+    OFFSET = 0x0  # No offset
+
+
+class HardNode(Node):
+    """
+    Hard node, where value = index + BIP32_HARDENED_CONSTANT
+    """
+    TAG = "H"  # "H" (or "'") means hard node (but use "H" for clarity)
+    OFFSET = 0x80000000  # 2**31, BIP32 "Hardening constant"
+
+
+def derive_child_key(
+    parent_key: bytes,
+    parent_chain_code: bytes,
+    node: Node,
+) -> Tuple[bytes, bytes]:
+    """
+    From BIP32:
+
+    The function CKDpriv((k_par, c_par), i) → (k_i, c_i) computes a child extended
+    private key from the parent extended private key:
+
+    - Check whether i ≥ 2**31 (whether the child is a hardened key).
+      - If so (hardened child):
+        let I = HMAC-SHA512(Key = c_par, Data = 0x00 || ser_256(kpar) || ser_32(i)).
+        (Note: The 0x00 pads the private key to make it 33 bytes long.)
+      - If not (normal child):
+        let I = HMAC-SHA512(Key = c_par, Data = ser_P(point(k_par)) || ser_32(i)).
+    - Split I into two 32-byte sequences, I_L and I_R.
+    - The returned child key k_i is parse_256(I_L) + k_par (mod n).
+    - The returned chain code c_i is I_R.
+    - In case parse_256(I_L) ≥ n or k_i = 0, the resulting key is invalid,
+      and one should proceed with the next value for i.
+      (Note: this has probability lower than 1 in 2**127.)
+    """
+    assert len(parent_chain_code) == 32
+    if isinstance(node, HardNode):
+        # NOTE Empty byte is added to align to SoftNode case
+        assert len(parent_key) == 32  # Should be guarenteed here in return statment
+        child = hmac_sha512(parent_chain_code, b"\x00" + parent_key + node.serialize())
+
+    else:
+        assert len(ec_point(parent_key)) == 33  # Should be guarenteed by Account class
+        child = hmac_sha512(parent_chain_code, ec_point(parent_key) + node.serialize())
+
+    assert len(child) == 64
+
+    if to_int(child[:32]) >= SECP256K1_N:
+        # Invalid key, compute using next node (< 2**-127 probability)
+        return derive_child_key(parent_key, parent_chain_code, node + 1)
+
+    child_key = (to_int(child[:32]) + to_int(parent_key)) % SECP256K1_N
+    if child_key == 0:
+        # Invalid key, compute using next node (< 2**-127 probability)
+        return derive_child_key(parent_key, parent_chain_code, node + 1)
+    return child_key.to_bytes(32, byteorder="big"), child[32:]
+
+
+class HDPath:
+    def __init__(self, path: str):
+        """
+        Constructor for this class. Initializes an hd account generator and
+        if possible the encoded key and the derivation path. If no arguments are
+        specified, create a new account with createAccount(...) or initialize
+        an account given a mnemonic and an optional password with initAccount(...)
```

Comment:
I don't think this comment is quite accurate, is it coming with the next PR?

Reply:
Oh yes, that comment is not accurate! Copy paste errors :grimacing: